### PR TITLE
backtester: Fix selling bug, add planned features list

### PR DESCRIPTION
**Pull request by gloriousCode**
# PR Description
@MadCozBadd noticed an error when the Backtester attempts to sell an asset, it unnecessarily shrinks the sell size in the function `reduceAmountToFitPortfolioLimit`. I've adjusted sell orders to simply not exceed the portfolio amount availible if that somehow were to occur.
I also neatened up an existing function, added a test and added some planned features to the backtester readme

## To replicate
- Use [this RSI strategy](https://gist.githubusercontent.com/gloriousCode/1b5689d6cb985d6a8204463512d9af3d/raw/eb85dd6f1ae1d8302745c2e8946b3e94a068c43a/rsi-test.strat)
- On branch master you will receive order warnings such as ` Order size shrunk from 0.17298078301363376 to 3.2159956297318508e-06 to remain within portfolio limits.`
- On this branch you will receive properly sized sell orders.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested
- [x] `go test ./... -race`
- [x] `golangci-lint run`
- [x] `TestVerifyOrderWithinLimits`
- [x]  Fixed test `TestExecuteOrderBuySellSizeLimit` as the sell fix highlighted bad test setup

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes

### CODE REVIEW

**Comment by mrzdvig** [comment]
Just hit the same issue. Waiting for the PR to be merged. Thanks.

**Review by shazbert — APPROVED** [review]
Tested and it does indeed sell now! ACK.

File: backtester/eventhandlers/exchange/exchange.go [hunk 1]
```diff
@@ -85,41 +85,27 @@ func (e *Exchange) ExecuteOrder(o order.Event, data data.Handler, bot *engine.En
 			return f, err
 		}
 	}
-	reducedAmount := reduceAmountToFitPortfolioLimit(adjustedPrice, amount, o.GetFunds())
-	if reducedAmount != amount {
-		f.AppendReason(fmt.Sprintf("Order size shrunk from %v to %v to remain within portfolio limits", amount, reducedAmount))
+
+	portfolioLimitedAmount := reduceAmountToFitPortfolioLimit(adjustedPrice, amount, o.GetFunds(), f.GetDirection())
+	if portfolioLimitedAmount != amount {
+		f.AppendReason(fmt.Sprintf("Order size shrunk from %v to %v to remain within portfolio limits", amount, portfolioLimitedAmount))
```
Comment (thrasher-) [inline backtester/eventhandlers/exchange/exchange.go:91]: Even though this request isn't related to this PR, please adjust these to %f to rid %v's display of scientific notation for small amounts and the same for the output under CanUseExchangeLimits

**Review by shazbert — APPROVED** [review]
Changes ACK.

**Review by thrasher- — APPROVED** [review]
tACK, master vs this PR!

image

**Pull request merged**